version: '3.8'

services:

  mysql:
    image: mysql:8
    container_name: es05-mysql-db
    restart: always
    environment:
      MYSQL_DATABASE: Applicazione
      MYSQL_USER: alnao
      MYSQL_PASSWORD: meraviglioso
      MYSQL_ROOT_PASSWORD: stupendo
    ports:
      - "3307:3306"
    volumes:
      - es05-mysql-data:/var/lib/mysql
#      - ./init/:/docker-entrypoint-initdb.d  # esegue script SQL all'avvio
      - ./init/users.sql:/docker-entrypoint-initdb.d/users.sql
    networks:
      - es05-app-network

  app:
    build: .
    container_name: es05-app
    restart: always
    ports:
      - "8045:8045"
    environment:
      #MYSQL_DATASOURCE_URL: jdbc:mysql://es05-mysql-db:3306/Applicazione
      MYSQL_DATASOURCE_PORT: 3306
      MYSQL_DATASOURCE_HOST: es05-mysql-db
      MYSQL_DATASOURCE_DATABASE: Applicazione
      MYSQL_DATASOURCE_USERNAME: alnao
      MYSQL_DATASOURCE_PASSWORD: meraviglioso
    depends_on:
      - mysql
    networks:
      - es05-app-network
    healthcheck:
      test: ["CMD-SHELL", "curl -f http://localhost:8045/actuator/health || exit 1"]
      interval: 30s
      timeout: 10s
      retries: 5

volumes:
  es05-mysql-data:

networks:
  es05-app-network:
